### PR TITLE
feat(extensions): add `--install-as` flag to `dfx extension install` command

### PR DESCRIPTION
**Pull request by smallstepman**
# Description

Continuation of https://github.com/dfinity/sdk/pull/2968. Allows to install extension under different name, which should help with naming conflicts resolution.

# How Has This Been Tested?

```console
❯ cargo run -- extension install sns --install-as snsns
   Compiling dfx v0.14.1 
    Finished dev [unoptimized + debuginfo] target(s) in 4.93s
     Running `target/debug/dfx extension install sns --install-as snsns`
  Extension 'sns' installed successfully, and is available as 'snsns'

❯ cargo run snsns                                   
    Finished dev [unoptimized + debuginfo] target(s) in 0.34s
     Running `target/debug/dfx snsns`
error: 'snsns' requires a subcommand but one was not provided
  [subcommands: config, deploy, import, download, help]

Usage: snsns [OPTIONS] <COMMAND>

For more information, try '--help'.
Error: Extension exited with non-zero status code '2'.
```

# Checklist:

- [x] The title of this PR complies with [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/).
- [ ] I have edited the CHANGELOG accordingly.
- [ ] I have made corresponding changes to the documentation.

### CODE REVIEW

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]
This feature seems like an unnecessary complication.  We can revisit in the future if the perceived utility exceeds "should help with".

**Comment by smallstepman** [comment]
Actually, there have been several occasions where I worked with extensions and wished this was already merged. I'd use that for installing `sns` extension, even when `dfx sns` command already exists (i.e. `dfx extension install sns --install-as sns-dev`) - very useful for testing things. I definitely don't see it as an unnecessary complication

**Comment by ericswanson-dfinity** [comment]
Okay, then let's try some DRY so that there will be fewer differences to maintain.

File: src/dfx/src/commands/extension/install.rs [hunk 1]
```diff
@@ -7,12 +7,26 @@ use clap::Parser;
...
     let mgr = env.new_extension_manager()?;
-    mgr.install_extension(&opts.name)?;
-    spinner.finish_with_message(format!("Extension '{}' installed successfully", opts.name).into());
+    mgr.install_extension(&opts.name, opts.install_as.as_deref())?;
+    spinner.finish_with_message(
+        format!(
+            "Extension '{}' installed successfully{}",
+            opts.name,
+            if opts.install_as.is_some() {
+                format!(", and is available as '{}'", opts.install_as.unwrap())
```
Comment (ericswanson-dfinity) [inline src/dfx/src/commands/extension/install.rs:24]: if let Some() to avoid unwrap

File: src/dfx/src/lib/extension/manager/install.rs [hunk 2]
```diff
@@ -73,29 +87,34 @@ impl ExtensionManager {
             .unpack(temp_dir.path())
             .map_err(|e| ExtensionError::DecompressFailed(download_url, e))?;
 
+        dbg!(std::fs::read_dir(temp_dir.path()).unwrap().count());
```
Comment (ericswanson-dfinity) [inline src/dfx/src/lib/extension/manager/install.rs:90]: intentional to leave this in?

**Review by ericswanson-dfinity — CHANGES_REQUESTED** [review]

**Review by ericswanson-dfinity — APPROVED** [review]

**Pull request merged**